analyzer/metadata-plugin/src/main/java/eu/fasten/analyzer/metadataplugin/MetadataDatabasePlugin.java

```diff
@@ -58,30 +59,35 @@ public MetadataDatabasePlugin(PluginWrapper wrapper) {
         private boolean restartTransaction = false;
         private final int transactionRestartLimit = 3;
 
-        /**
-         * Constructor for MetadataPlugin with default DSLContext
-         * with parameters from `postgres.properties`.
-         *
-         * @param callgraphTopic Topic from which to consume call graphs
-         * @throws IOException              if cannot read file `postgres.properties`
-         * @throws SQLException             if cannot connect to the database
-         * @throws IllegalArgumentException if database URL in `postgres.properties` is malformed
-         */
-        public MetadataDBExtension(String callgraphTopic) throws IOException, SQLException,
-                IllegalArgumentException {
-            this(callgraphTopic, PostgresConnector.getDSLContext());
-        }
+//        /**
+//         * Constructor for MetadataPlugin with default DSLContext
+//         * with parameters from `postgres.properties`.
+//         *
+//         * @param callgraphTopic Topic from which to consume call graphs
+//         * @throws IOException              if cannot read file `postgres.properties`
+//         * @throws SQLException             if cannot connect to the database
+//         * @throws IllegalArgumentException if database URL in `postgres.properties` is malformed
+//         */
+//        public MetadataDBExtension(String callgraphTopic) throws IOException, SQLException,
+//                IllegalArgumentException {
+//            this(callgraphTopic, PostgresConnector.getDSLContext());
+//        }
 
-        /**
-         * Constructor for MetadataPlugin with provided DSLContext.
-         *
-         * @param callgraphTopic Topic from which to consume call graphs
-         * @param dslContext     DSLContext for jOOQ to query the database
-         */
-        public MetadataDBExtension(String callgraphTopic, DSLContext dslContext) {
-            super();
-            this.dslContext = dslContext;
-            this.topic = callgraphTopic;
+//        /**
+//         * Constructor for MetadataPlugin with provided DSLContext.
+//         *
+//         * @param callgraphTopic Topic from which to consume call graphs
+//         * @param dslContext     DSLContext for jOOQ to query the database
+//         */
+//        public MetadataDBExtension(String callgraphTopic, DSLContext dslContext) {
+//            super();
+//            this.dslContext = dslContext;
+//            this.topic = callgraphTopic;
+//        }
+
+        @Override
+        public void getDBAccess(String DBUrl, String username, String password) throws SQLException {
+            PostgresConnector.getDSLContext(DBUrl, username, password);
```

Comment:
You should assign it to `dslContext` here. Otherwise, you don't really get a DSLContext required to interract with database. It should be like this:
```java
this.dslContext = PostgresConnector.getDSLContext(DBUrl, username, password);
```
And then it would make a bit more sense to call this method something like `setupDBAccess()` or simply `connectDB()`

Reply:
Okay. Thanks. I will fix it.